src/OFS/DTMLMethod.py

```diff
@@ -266,10 +266,12 @@ def manage_upload(self, file='', REQUEST=None):
         if self.wl_isLocked():
             raise ResourceLockedError('This DTML Method is locked.')
 
-        if not isinstance(file, binary_type):
+        if not isinstance(file, basestring):
```

Comment:
`basestring` is not defined in Python 3

Reply:
@davisagli You are right, but someone defined it in the this Python module, see: https://github.com/zopefoundation/Zope/blob/bc180d5f2c2ae6099f888e1f8b4e520dbe6b46f9/src/OFS/DTMLMethod.py#L46

Reply:
That `if sys.version_info` could use the PY3 symbol imported above

Reply:
Although it is not completely the same it is close enough, as we will drop Python 2 support before Python 4 comes out. :)